# Selection for binders *in vitro*

In the case of selection *in vivo*, genotype (plasmid or genomic trait) is naturally linked with the phenotype in the cell. Selection *in vitro* necessitates the creation of such link. A multitude of such selection methods have been developed where selected molecules are physically linked to its gene:

- Via translating ribosome in the [ribosome display](ribodisplay.md)
- Via puromycin linker in the [mRNA and cDNA display](mRNAdisplay.md)
- Via an auxiliary protein RepA in [CIS display](cisdisplay.md)
- Being entrapped in the vesicle as in [compartmentalized self-replication (CSR)](CSR_SBL.md)
- Being entrapped in the vesicle and the bead as in [Compartmentalized bead labelling (SBL)](CSR_SBL.md)

The *in vitro* selection methods have the following advantages:

- Libraries with up to 10^15 mutants can be selected (except in those where DNA is entrapped in vesicles, where this diversity is down to 10^9 as only a single DNA molecule per vesicle is allowed)
- *in vitro* amplification conveniently integrate random mutagenesis into the selection procedure
- The outcomes may be least biased in copmarison to *in vivo* methods where selection is affected by the physiological state of the cell
- Selection of unnatural polymers which cannot be replicated in cells such as XNAs is possible.
 
With that, except for CSR and SBL protocols, all these methods are typically used in selecting binders. And as far as literature goes, only rarely used in the selection of enzymatic activities.
