# Recombinatorial gene and domain shuffling

## Introduction
DNA shuffling is an established recombinatorial method that was originally developed to increase the speed of directed evolution experiments beyond what could be accomplished using error-prone PCR alone. To achieve this, mutated copies of a protein-coding sequence are fragmented with DNase I or restriction enzyme and the fragments are then reassembled in a PCR without primers. 

## In Vitro Homologous Recombination

DNA shuffling was the first reported method of *in vitro* recombination. DNAse I is first used to randomly fragment the parental genes. Then, provided there is sufficient sequence similarity between overlapping regions of DNA fragments from different parents, the fragments can anneal to one another and reassemble into a full-length gene using a primer-less PCR. Depending on the total gene size, the degree of sequence similarity between parental DNA strands, and desired number of crossovers, fragment sizes after DNAse I treatment can vary from as low as 10–50 bp to greater than 1 kb. 

Random point mutations tend to occur at low rates during recombination even with a high-fidelity polymerase, and researchers will often intentionally employ error-prone PCR during PCR-based gene recombination to further diversify their library.

DNA shuffling can aslso be done on a set of naturally occuring homologous genes, instead of generted by epPCR a set of mutant genes. This technique is called “family shuffling”. A number of other PCR-based and homology-dependent [protocols](https://link.springer.com/protocol/10.1385/1-59259-194-9:243) have since been developed which accomplish essentially the same as DNA shuffling.

Examples include the staggered extension process, random-priming *in vitro* recombination, and random chimeragenesis on transient templates. In general these methods work well for DNA sequences sharing at least 65% identity, although higher sequence identity more readily yields a greater number of crossovers.

*In vitro* recombination methods are also often used in directed evolution, even when the only genetic diversity is introduced by random mutagenesis of a single parent gene. Here, one or more rounds of mutagenesis and screening to isolate improved variants results in a handful of mutant genes, each carrying a different set of point mutations. By shuffling these highly identical mutant DNA sequences, one can readily obtain a library containing all combinations of point mutations. Beneficial mutations can be combined and may show additive effects, while any potentially deleterious mutations that have accumulated will be eliminated by ‘back-crossing’ with the wild-type sequence.


