# Introduction

In vitro compartmentalisation (IVC) mimics cells by entrapping the genotype (usually in the form of DNA), the in vitro transcription-translation system, and the newly translated protein in a water-in-oil emulsion. The first report of this method dates back to [1996](https://www.nature.com/articles/nbt0798-652). Since then, a wide variety of IVC protocols has been developed (see below). Correct mixing of oil, detergents, reaction components, and DNA ensures that each droplet contains only a single copy of the genotype or none at all. The maximum number of mutants that can be reliably screened with IVC rarely exceeds 10<sup>9</sup>. Otherwise, the IVC method is far superior to other in vitro selection protocols. It allows the selection of binders and enzymes since the product of their activity is also trapped in the vesicle.

# Principle and Examples

Different versions of covalent DNA display, where each expressed protein gets physically connected to its encoding DNA, are also examples of IVC [here](https://academic.oup.com/peds/article/20/2/57/2270161)and[here]](https://pubmed.ncbi.nlm.nih.gov/17948318/).

A powerful method that follows the notion of entrapping genotype and phenotype in the same droplet was used in Compartmentalised Self-Replication or [CSR](https://pubmed.ncbi.nlm.nih.gov/30040233/). In this protocol, cells expressing mutant DNA polymerases get entrapped in droplets, along with primers for amplification of the gene encoding the DNA polymerase, nucleotides, and reaction buffer. Following lysis of the cell, both DNA polymerase mutant and the encoding plasmid are released. The DNA of the most active DNA polymerase variants gets exponentially amplified in the emulsion to be used in further rounds of CSR.  

In another development of IVC, droplets contain beads, which are coupled to both DNA and a product of the reaction that is used as a readout in the subsequent [flow cytometry sorting](https://febs.onlinelibrary.wiley.com/doi/full/10.1016/S0014-5793%2802%2903740-7). A version of this protocol suitable for selecting reverse transcriptases of both natural and xenonucleic acids (Compartmentalized Bead Labelling or [CBL](https://www.nature.com/articles/s41557-020-0502-8)) demonstrates the huge potential that IVS has over other in vitro selection methods.

# Challenges

Different versions of IVC have their inherent limits that are not addressed here. Challenges in setting up a version of IVC that contain in vitro transcription-translation systems often come from two sources:

- Vesicle formation is a difficult procedure that includes the use of detergent and high-frequency mixing. Both factors can irreversibly damage translation components. Thus, known protocols should be followed with great attention to the details. Expect lots of optimisation initially.
- A degree in polydispersity among droplets is another challenge, as it leads to variation in the reaction rates due to the dilution effect. Microfluidic setups can be used for the preparation of monodisperse droplets. However, these setups have orders of magnitude lower levels of throughput.

# Future directions

As can be appreciated from the number of exciting developments of IVC, this method has tremendous potential for further development. As it is ideally suited for the selection of enzymatic activities, lots of studies are expected in this direction, especially in the field of nucleic acid enzymes. In principle, it can be used in the selection of systems containing several molecular components, the combined activity of which leads to the production of a marker for a readout. Entrapping activity components in droplets have a lot in common with artificial cells. Thus the latter will likely be a powerful method in both the directed evolution of whole artificial cells as well as their functional parts.

# Links 

Good methodological article
https://www.nature.com/articles/nmeth897

Comprehensive Review
https://pubs.rsc.org/en/content/articlehtml/2020/cs/c8cs00981c









